LiteCore/Support/FilePath.cc

```diff
@@ -80,13 +81,30 @@ static int copyfile(const char* from, const char* to)
         errno = e;
         return write_fd;
     }
-    
-    if(sendfile(write_fd, read_fd, &offset, stat_buf.st_size) < 0) {
-        int e = errno;
-        close(read_fd);
-        close(write_fd);
-        errno = e;
-        return -1;
+
+    size_t expected = stat_buf.st_size;
+    ssize_t bytes = 0;
+    int noProgressRetry = 2;
+    while (bytes < expected) {
+        expected -= bytes;
+        bytes = sendfile(write_fd, read_fd, &offset, expected);
+        if (bytes < 0) {
+            int e = errno;
+            close(read_fd);
+            close(write_fd);
+            errno = e;
+            return -1;
+        } else if (bytes == 0) {
```

Comment:
This might be overdesign. If sendfile can't copy any bytes I would consider it an error and give up, even if it didn't return an error.

Reply:
I agreed. I changed it in porting it to lithium,
        } else if (bytes == 0) {
            // zero bytes are read. Do we want to try again? Well, let's consider it as an error
            Warn("sys/sendfile makes no progress copying %s to %s and we bail out as failure.", from, to);
       ...
I will add you to the reviewer in the lithium PR.